youtube_dl/extractor/generic.py

```diff
@@ -1817,6 +1817,11 @@ def filter_video(urls):
                     'url': new_url,
                 }
         if not found:
+            # Last resort: just look for anything that looks like a media URL
+            found = set(re.findall(
+                r'\b((?:https?:)?//[A-Za-z0-9.:[\]]+/[A-Za-z0-9\-\._~:/()*+,;=%]+\.(?:mp3|mp4|flv|m3u8|smil)(?:\?[A-Za-z0-9\-\._~:/[\]()*+,;=%]*)?)\b',
```

Comment:
As you have pointed, it could match unwanted urls so I would avoid adding it. If you are only interested in ComCarrCoff, you could try to add a custom extractor because the method may not be used in too many sites..